src/Encoding.chpl

```diff
@@ -0,0 +1,248 @@
+/* Documentation for Encoding */
+module Encoding {
```

Comment:
I'm not confident than Encoding will remain the name of this module but I recommend that we leave that choice for the API review. (In particular I think it probably should mention protobuf somewhere in the name).